python/ray/tests/test_runtime_env_profiler.py

```diff
@@ -0,0 +1,207 @@
+            def run(self):
+                return 0
+
+        ray_actor = NsightActor.remote()
+        ray.get(
+            [
+                test_generate_report.remote(),
+                test_generate_report.remote(),
+                ray_actor.run.remote(),
+                ray_actor.run.remote(),
+            ]
+        )
+        ray.kill(ray_actor)
+
+        wait_for_condition(wait_for_report, logs_dir=logs_dir, num_reports=3)
+        nsys_reports = glob.glob(os.path.join(f"{logs_dir}/*.nsys-rep"))
+        assert len(nsys_reports) == 3  # ray actor used only one worker process
+        for report in nsys_reports:
+            assert "ray_task" in report or "ray_actor" in report
+            os.remove(report)
+
+    @pytest.mark.skipif(
+        os.environ.get("CI") and sys.platform != "linux",
+        reason="Requires PR wheels built in CI, so only run on linux CI machines.",
+    )
+    def test_nsight_invalid_option(
+        self,
+        shutdown_only,
+    ):
+        """Test Nsight profile for invalid option"""
+        ray.init()
+        session_dir = ray.worker._global_node.get_session_dir_path()
+        logs_dir = f"{session_dir}/logs/nsight/"
+
+        # test nsight default config
+        @ray.remote(
+            runtime_env={
+                "nsight": {
+                    "-not-option": "random",
+                }
+            }
+        )
+        def test_invalid_nsight():
+            return 0
+
+        with pytest.raises(
+            RuntimeEnvSetupError,
+            match="nsight profile failed to run with the following error message",
+        ):
+            ray.get(test_invalid_nsight.remote())
+
+        nsys_reports = glob.glob(os.path.join(f"{logs_dir}/*.nsys-rep"))
+        assert len(nsys_reports) == 0
+
+        """Test Nsight profile for unavailable string option"""
+
+        # test nsight default config
+        @ray.remote(runtime_env={"nsight": "not_default"})
+        def test_wrong_config_nsight():
+            return 0
+
+        with pytest.raises(
+            RuntimeEnvSetupError, match="Unsupported nsight config: not_default."
+        ):
+            ray.get(test_wrong_config_nsight.remote())
+
+        nsys_reports = glob.glob(os.path.join(f"{logs_dir}/*.nsys-rep"))
+        assert len(nsys_reports) == 0
+
+    @pytest.mark.skipif(
+        sys.platform == "linux",
+        reason="For non linux system, Nsight CLI is not supported",
+    )
+    def test_nsight_unsupported(
+        self,
+        shutdown_only,
+    ):
+        """Test Nsight profile for unsupported OS"""
+        ray.init()
+        session_dir = ray.worker._global_node.get_session_dir_path()
+        logs_dir = f"{session_dir}/logs/nsight/"
+
+        # test nsight default config
+        @ray.remote(runtime_env={"nsight": "default"})
+        def test_nsight_not_supported():
+            return 0
+
+        with pytest.raises(
+            RuntimeEnvSetupError, match="Nsight CLI is only available in Linux."
```

Comment:
Not just this ,but can you test if the right error (RunttimeEnvSetupError) is raised when nsight cli is not downloaded?